src/components/RoadmapList/RoadmapList.tsx

```diff
@@ -2,56 +2,96 @@ import { useEffect, useState } from 'react';
 import ReactPaginate from 'react-paginate';
 import { Link } from 'react-router-dom';
 
-import { getRoadmapList } from 'src/api';
+import {
+  getFiltredList,
+  getFiltredListWithSearch,
+  getRoadmapList,
+} from 'src/api';
 import { ROUTES } from 'src/routing/routes';
 
 import Button, { buttonVariant } from 'components/ui/Button';
 import Container from 'components/ui/Container';
 import Icon from 'components/ui/Icon';
 
+import Filters from './Filters';
 import { RoadmapListItem } from './type';
 
 import style from './RoadmapList.module.css';
 
-export const ROADMAPS_PER_PAGE = 6;
+const ROADMAPS_PER_PAGE = 6;
+const DEFAULT_FILTER_VALUES = {
+  category: '',
+  level: '',
+  domain: '',
+  search: '',
+};
 
 const RoadmapList = () => {
   const [currentPage, setCurrentPage] = useState(1);
   const [totalPage, setTotalPage] = useState(1);
   const [list, setList] = useState<RoadmapListItem[]>([]);
+  const [filters, setFilters] = useState(DEFAULT_FILTER_VALUES);
 
   useEffect(() => {
-    getRoadmapList(currentPage).then(({ data }) => {
-      setList(data.results);
-      setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
-    });
-  }, [currentPage]);
+    if (filters === DEFAULT_FILTER_VALUES) {
+      getRoadmapList(currentPage).then(({ data }) => {
+        setList(data.results);
+        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+      });
+    }
+  }, [currentPage, filters]);
 
   const handlePageChange = ({ selected }: { selected: number }) => {
     setCurrentPage(selected + 1);
+    if (filters !== DEFAULT_FILTER_VALUES) {
+      handleFiltration(selected + 1)
+    }
+  };
+
+  const handleSelect = (name: string, value: string) => {
+    setFilters((state) => ({
+      ...state,
+      [name]: value,
+    }));
+  };
+
+  const handleSearch = () => {
+    handleFiltration()
+  };
+
+  const handleFiltration = (currentPage = 1)=>{
+    if (filters.search) {
+      getFiltredListWithSearch(filters, currentPage).then(({ data }) => {
+        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+        setList(data.results);
+      });
+      return;
+    }
+    if (!filters.search) {
+      getFiltredList(filters, currentPage).then(({ data }) => {
+        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
+        setList(data.results);
+      });
+      return;
+    }
```

Comment:
({ data }) => {
        setTotalPage(Math.ceil(data.count / ROADMAPS_PER_PAGE));
        setList(data.results);
      })
      
      It can be relocated in func and used here as callback instead
      Also I think it's more clear to use async/await

Reply:
yes we have the same functions in then, and 2 return looks not good. Use if else instead